src/buffer.rs

```diff
@@ -225,6 +225,17 @@ impl AudioBuffer {
         self.channel_data(channel_number).as_slice()
     }
 
+    /// Return a mutable slice of the underlying data of the channel
+    ///
+    /// # Panics
+    ///
+    /// This function will panic if:
+    /// - the given channel number is greater than or equal to the given number of channels.
+    pub fn get_channel_data_mut(&mut self, channel_number: usize) -> &mut [f32] {
```

Comment:
Take care! I slipped in a new public method here. Similar to #214 I ran into an issue where I could not port the example directly.
The method is not part of the spec, but it is idiomatic rust (compare `get_channel_data`/`get_channel_data_mut` to HashMap `get`/`get_mut`).
What do you think?

Reply:
Not sure this would solve #214 though as you can't change the buffer length from there, no?

But it feels quite sensible and handy to me

Reply:
Yeah that is true of course.
This makes me think, what if you use the JS code from #214 and create a stereo buffer where both channels have unequal length? insanity

Reply:
Yup... JS you know :)